src/super_gradients/training/models/segmentation_models/stdc.py

```diff
@@ -0,0 +1,646 @@
+        attributes_to_delete = ["aux_head_s16", "aux_head_s32", "detail_head8"]
+        for attr in attributes_to_delete:
+            if hasattr(self, attr):
+                delattr(self, attr)
+
+    @property
+    def use_aux_heads(self):
+        return self._use_aux_heads
+
+    @use_aux_heads.setter
+    def use_aux_heads(self, use_aux: bool):
+        """
+        private setter for self._use_aux_heads, called every time an assignment to self._use_aux_heads is applied.
+        if use_aux is False, `_remove_auxiliary_and_detail_heads` is called to delete auxiliary and detail heads.
+        if use_aux is True, and self._use_aux_heads was already set to False a ValueError is raised, recreating
+            aux and detail heads outside init method is not allowed, and the module should be recreated.
+        """
+        if use_aux is True and self._use_aux_heads is False:
+            raise ValueError("Cant turn use_aux_heads from False to True, you should initiate the module again with"
+                             " `use_aux_heads=True`")
+        if not use_aux:
+            self._remove_auxiliary_and_detail_heads()
+        self.cp.use_aux_heads = use_aux
+        self._use_aux_heads = use_aux
+
+    @property
+    def backbone(self):
+        """
+        For SgModel load_backbone compatibility.
+        """
+        return self.cp.backbone
+
+    def init_params(self):
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(m.weight, mode='fan_out', nonlinearity="relu")
+                if m.bias is not None:
+                    nn.init.constant_(m.bias, 0)
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.constant_(m.weight, 1)
+                nn.init.constant_(m.bias, 0)
+            elif isinstance(m, nn.Linear):
+                nn.init.normal_(m.weight, std=0.001)
+                if m.bias is not None:
+                    nn.init.constant_(m.bias, 0)
+
+    def forward(self, x):
+        cp_outs = self.cp(x)
+        feat8, feat_cp8 = cp_outs[0], cp_outs[1]
+        # fuse stage 3 with result of context path after ARM modules.
+        feat_out = self.ffm(spatial_feats=feat8, context_feats=feat_cp8)
+        feat_out = self.segmentation_head(feat_out)
+
+        if not self.use_aux_heads:
+            return feat_out
+        feat16, feat32 = cp_outs[2], cp_outs[3]
+        detail_out8 = self.detail_head8(feat8)
+
+        aux_out_s16 = self.aux_head_s16(feat16)
+        aux_out_s32 = self.aux_head_s32(feat32)
+
+        return feat_out, aux_out_s32, aux_out_s16, detail_out8
+
+    def replace_head(self, new_num_classes: int, **kwargs):
+        ffm_channels = self.ffm.attention_block[-2].out_channels
+        dropout = self.segmentation_head[0].seg_head[1].p
+
+        self.segmentation_head = nn.Sequential(
+            SegmentationHead(ffm_channels, ffm_channels, new_num_classes, dropout=dropout),
```

Comment:
you can replace only the first element in the head sequentials

Reply:
done